Source/Shaders/GlobeFS.glsl

```diff
@@ -110,6 +116,25 @@ vec4 sampleAndBlend(
     }
 #endif
 
+    if(applyTextureColorPalette > 0.0) {
+        float step = 1.0/1024.0;
+        float a1 = texture2D(texture, textureCoordinates).r;
+        float a2 = texture2D(texture, textureCoordinates + vec2(step, 0.0)).r;
+        float a3 = texture2D(texture, textureCoordinates + vec2(0.0, step)).r;
+        float a4 = texture2D(texture, textureCoordinates + vec2(step, step)).r;
+        if ((a1 != 0.0 && (a2 == 0.0 || a3 == 0.0 || a4 == 0.0)) || a1 == 0.0) {
+        } else {
+          vec2 f = fract(textureCoordinates * vec2(1024.0, 1024.0));
+          float tA = mix(a1, a2, f.x);
+          float tB = mix(a3, a4, f.x);
+          a1 = mix(tA, tB, f.y);
+        }
+
+        vec4 pixColor = texture2D(textureColorPalette, vec2(0.0, a1));
```

Comment:
Using u_dayTextureColorPalette[index] here results in 

`'[]' : Index expression must be constant`